src/sas/sasgui/perspectives/fitting/basepage.py

```diff
@@ -1690,7 +1703,38 @@ def _draw_model(self, update_chisqr=True, source='model'):
 
         :param chisqr: update chisqr value [bool]
         """
-        wx.CallAfter(self._draw_model_after, update_chisqr, source)
+        # Get the time
+        current_time = time.time()
+
+        # When loading we slam a number of fits through here
+        # let's filter these out to start with
+        if current_time > (self.last_time_fit_submitted + 0.1):
+            # Submitting the rest
+            self.threaded_draw_queue.put([update_chisqr, source])
+        else:
+            pass
+
+        self.last_time_fit_submitted = current_time
+
+    def _threaded_draw_worker(self, threaded_draw_queue):
+        while True:
+            # Check to see is a manager is running and a calc is running
+            if ((self._manager.calc_1D is not None) and (self._manager.calc_1D.\
+                isrunning() == True)) or ((self._manager.calc_2D is not None)\
```

Comment:
no need for `(self._manager.calc_1D.isrunning == True)` can just be `(self._manager.calc_1D.isrunning())`